images/alpine/cloudbuild.yaml

```diff
@@ -1,19 +1,40 @@
 steps:
-  - name: gcr.io/cloud-builders/docker
-    args:
-    - build
-    - -t
-    - gcr.io/$PROJECT_ID/alpine:$_GIT_TAG
-    - --build-arg=IMAGE_ARG=gcr.io/$PROJECT_ID/alpine:$_GIT_TAG
-    - .
+# Set up multiarch support.
+  - name: 'gcr.io/cloud-builders/docker'
+    args: ['run', '--rm', '--privileged', 'multiarch/qemu-user-static', '--reset', '-p', 'yes']
+    id: 'qemu'
+    waitFor: ['-']
+
+# Buildx builder.
+  - name: 'gcr.io/cloud-builders/docker'
+    id: 'buildx-builder'
+    env:
+      - 'DOCKER_CLI_EXPERIMENTAL=enabled'
+    args: [ 'buildx', 'create', '--use', '--name=k8s-testimages']
+    waitFor: ['qemu']
+
+# Buildx.
+  - name: 'gcr.io/cloud-builders/docker'
+    id: 'buildx-images-tag'
+    env:
+      - 'DOCKER_CLI_EXPERIMENTAL=enabled'
+    args: ['buildx', 'build', '--platform=linux/amd64,linux/arm64,linux/ppc64le', '--push',
+    '-t', 'gcr.io/$PROJECT_ID/alpine:$_GIT_TAG',
+    '--build-arg', 'IMAGE_ARG=gcr.io/$PROJECT_ID/alpine:$_GIT_TAG',
+    '--pull', '.']
     dir: .
-  - name: gcr.io/cloud-builders/docker
-    args:
-    - tag
-    - gcr.io/$PROJECT_ID/alpine:$_GIT_TAG
-    - gcr.io/$PROJECT_ID/alpine:latest
+    waitFor: ['buildx-builder']
+
+  - name: 'gcr.io/cloud-builders/docker'
+    id: 'buildx-images-latest'
+    env:
+      - 'DOCKER_CLI_EXPERIMENTAL=enabled'
+    args: ['buildx', 'build', '--platform=linux/amd64,linux/arm64,linux/ppc64le', '--push',
+    '-t', 'gcr.io/$PROJECT_ID/alpine:latest',
+    '--build-arg', 'IMAGE_ARG=gcr.io/$PROJECT_ID/alpine:latest',
+    '--pull', '.']
+    dir: .
+    waitFor: ['buildx-images-tag']
+
 substitutions:
   _GIT_TAG: '12345'
-images:
-  - 'gcr.io/$PROJECT_ID/alpine:$_GIT_TAG'
```

Comment:
Please preserve the expected images to publish.

Reply:
Hi,
In step 3, those images with fat-manifest will be pushed into registry with the 'buildx --push'.
I have tested the build methods(make & prowjob) with my private gcloud account, the multi-arch images can be pushed into my private gcloud registry automatically.

On the other hand, we can't load the multi-arch images into the local docker daemon with the 'buildx --load'.
Please see the issue as reference:
https://github.com/docker/buildx/issues/59
If we keep this images field, the build will report an error

Reply:
Thanks, this is my first experience with buildx and multiarch images (thanks so much for working on this).

And if the push fails then the build fails? How about then adding a comment at the bottom about what images we expect this to push and why we don't use cloud build's auto verification?

Reply:
> And if the push fails then the build fails? 
Yes.
> How about then adding a comment at the bottom about what images we expect this to push and why we don't use cloud build's auto verification?
The comment has been added.